src/zbytes.rs

```diff
@@ -135,8 +125,33 @@ pub unsafe extern "C" fn z_bytes_to_slice(
     dst: &mut MaybeUninit<z_owned_slice_t>,
 ) -> z_result_t {
     let payload = this.as_rust_type_ref();
-    dst.as_rust_type_mut_uninit()
-        .write(payload.to_bytes().into_owned().into());
+    match payload.to_bytes() {
+        std::borrow::Cow::Borrowed(s) => {
+            // force a copy if zbytes is a single shared memory slice
+            #[cfg(all(feature = "unstable", feature = "shared-memory"))]
+            if payload.as_shm().is_some() {
+                dst.as_rust_type_mut_uninit().write(s.to_owned().into());
+                return result::Z_OK;
+            }
+            // otherwise we avoid a copy and instead just place zbytes clone into an owned_slice context to ensure
+            // that it will outlive the pointed data
+            let context = Box::leak(Box::new(payload.clone())) as *mut ZBytes as *mut _;
```

Comment:
Wouldn't it be more efficient to use `std::mem::forget` to avoid running the destructor and use `*mut ZBytes` as a context, instead of relying on ownership semantics of `Box`?

Reply:
As far as I understand mem::forget only does not run the drop, but it will not prevent rust from reusing memory occupied by payload.clone() for other purposes.